apple/OmnivoreKit/Sources/App/PDFSupport/PDFViewerViewModel.swift

```diff
@@ -18,25 +18,15 @@ public final class PDFViewerViewModel: ObservableObject {
   }
 
   public func dataURL(remoteURL: URL) -> URL {
-    if let storedURL = storedURL {
+    // TODO: we should probably not reach this point of localPdfURL is not set
```

Comment:
I think we could get here when the user taps a cell where prefetch is still in process. Or would this cell not even show up until the pdf data is stored on device?

Reply:
@satindar ah yeah, I couldn't figure out how to handle this, but we want it to be like fetching HTML content. We just show the spinner until the full PDF is downloaded.